substrate/frame/revive/src/evm/block_hash/block_builder.rs

```diff
@@ -25,56 +25,80 @@ use crate::{
 		},
 		Block, HashesOrTransactionInfos, TYPE_EIP1559, TYPE_EIP2930, TYPE_EIP4844, TYPE_EIP7702,
 	},
-	ReceiptGasInfo,
+	Config, Pallet, ReceiptGasInfo,
 };
 
 use alloc::{vec, vec::Vec};
 
 use codec::{Decode, Encode};
-use frame_support::DefaultNoBound;
+use frame_support::traits::Time;
 use scale_info::TypeInfo;
+use sp_arithmetic::traits::Saturating;
 use sp_core::{keccak_256, H160, H256, U256};
+use sp_runtime::traits::{One, Zero};
 
 const LOG_TARGET: &str = "runtime::revive::block_builder";
 
 /// Ethereum block builder designed to incrementally build the transaction and receipt trie roots.
 ///
 /// This builder is optimized to minimize memory usage and pallet storage by leveraging the internal
 /// structure of the Ethereum trie and the RLP encoding of receipts.
-#[derive(DefaultNoBound)]
 pub struct EthereumBlockBuilder<T> {
 	pub(crate) transaction_root_builder: IncrementalHashBuilder,
 	pub(crate) receipts_root_builder: IncrementalHashBuilder,
 	pub(crate) tx_hashes: Vec<H256>,
 	gas_used: U256,
+	base_fee_per_gas: U256,
+	block_gas_limit: U256,
 	logs_bloom: LogsBloom,
 	gas_info: Vec<ReceiptGasInfo>,
 	_phantom: core::marker::PhantomData<T>,
 }
 
+impl<T: Config> Default for EthereumBlockBuilder<T> {
```

Comment:
dq: In pallet revive we should only allow constructing the `EthereumBlockBuilder` via `from_ir` method. Did we add Default for consistency with IR counterpart?